java/flight/flight-jdbc-driver/src/test/java/org/apache/arrow/driver/jdbc/FlightCookieTest.java

```diff
@@ -0,0 +1,130 @@
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ErrorCollector;
+
+import com.google.protobuf.Message;
+
+public class FlightCookieTest {
+
+  private static final MockFlightSqlProducer FLIGHT_SQL_PRODUCER = new MockFlightSqlProducer();
+  private static final String REGULAR_QUERY_SAMPLE = "SELECT * FROM NOT_UPDATE_QUERY";
+  private static final Schema REGULAR_QUERY_SCHEMA =
+      new Schema(
+          Collections.singletonList(Field.nullable("placeholder", Types.MinorType.VARCHAR.getType())));
+  private static final int ROW_COUNT = 10;
+
+  @Rule
+  public final ErrorCollector collector = new ErrorCollector();
+
+  @ClassRule
+  public static final FlightServerTestRule FLIGHT_SERVER_TEST_RULE =
+      FlightServerTestRule.createStandardTestRule(FLIGHT_SQL_PRODUCER);
+
+  @BeforeClass
+  public static void setup() {
+    FLIGHT_SQL_PRODUCER.addSelectQuery(
+        REGULAR_QUERY_SAMPLE,
+        REGULAR_QUERY_SCHEMA,
+        Collections.singletonList(listener -> {
+          try (final BufferAllocator allocator = new RootAllocator(Long.MAX_VALUE);
+               final VectorSchemaRoot root = VectorSchemaRoot.create(REGULAR_QUERY_SCHEMA,
+                   allocator)) {
+            listener.start(root);
+            listener.putNext();
+          } catch (final Throwable throwable) {
+            listener.error(throwable);
+          } finally {
+            listener.completed();
+          }
+        }));
+    final Message commandGetTableTypes = FlightSql.CommandGetTableTypes.getDefaultInstance();
+    final Consumer<FlightProducer.ServerStreamListener> commandGetTableTypesResultProducer = listener -> {
+      try (final BufferAllocator allocator = new RootAllocator();
+           final VectorSchemaRoot root = VectorSchemaRoot.create(FlightSqlProducer.Schemas.GET_TABLE_TYPES_SCHEMA,
+               allocator)) {
+        final VarCharVector tableType = (VarCharVector) root.getVector("table_type");
+        range(0, ROW_COUNT).forEach(
+            i -> tableType.setSafe(i, new Text(format("table_type #%d", i))));
+        root.setRowCount(ROW_COUNT);
+        listener.start(root);
+        listener.putNext();
+      } catch (final Throwable throwable) {
+        listener.error(throwable);
+      } finally {
+        listener.completed();
+      }
+    };
+    FLIGHT_SQL_PRODUCER.addCatalogQuery(commandGetTableTypes, commandGetTableTypesResultProducer);
+  }
+
+  @Test
+  public void testeCookies() throws SQLException {
+    try (Connection connection = FLIGHT_SERVER_TEST_RULE.getConnection()) {
+
+      // Since the first cookie was just updated by the client it was not send yet.
+      // That's why we're checking for null.
+      Assert.assertNull(FLIGHT_SERVER_TEST_RULE.getFactory().getCookie());
+
+      // Run another action for check if the cookies was sent by the client.
+      Statement statement = connection.createStatement();
```

Comment:
try with resources for both the Statement and the Catalogs ResultSet. If the test fails, the close() calls won't happen.

Reply:
fixed